test/extensions/filters/http/ratelimit/ratelimit_test.cc

```diff
@@ -1489,6 +1494,47 @@ TEST_F(HttpRateLimitFilterTest, IgnoreVHRateLimitOptionWithOutRouteRateLimit) {
       0, filter_callbacks_.clusterInfo()->statsScope().counterFromStatName(ratelimit_ok_).value());
 }
 
+// Tests that the domain is properly overridden when set at the per-route level
+TEST_F(HttpRateLimitFilterTest, PerRouteDomainSet) {
+  SetUpTest(filter_config_);
+  envoy::extensions::filters::http::ratelimit::v3::RateLimitPerRoute settings;
+  settings.set_domain("bar");
```

Comment:
nit: can we have a local_variable like `const std::string per_route_domain = "bar"` And then, set it here and use it in limit() function at line#1515.

This is to highlight the check of per route domain name override, which is the key of this test.